lib/addMethod/addMethodFunction.js

```diff
@@ -0,0 +1,56 @@
+var _ = require('lodash');
+var when = require('when');
+var logger = require('../logger');
+
+var globalize = require('./globalize');
+
+//The format of the response should always be an object with headers and body
+function formatResponse (headers, body) {
+	return {
+		headers: ( _.isPlainObject(headers) ? headers : {} ),
```

Comment:
as a general note, it's good to be more explicit what is allowed as `headers` and throw an error rather than swallowing it and replacing with {}

Reply:
otherwise it's surprising when something does not work, as there is no feedback